### PR TITLE
feat(Drawer): update stacked Drawer layout

### PR DESCRIPTION
**Pull request by vyshkov**
**What is the problem this PR is trying to solve?**

Frontify definition for the Drawer component updated http://guidelines.talend.com/document/92132#/navigation-layout/drawer-br

We need to center drawer content like on the picture:
image

**What is the chosen solution to this problem?**

Updated styles and layout

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

# Breaking change
If you have wide drawer content (like multi-column forms) you will need to update them.

Here an example of TMC task https://jira.talendforge.org/browse/TMC-11183

### CODE REVIEW

File: output/components.sasslint.txt [hunk 1]
```diff
@@ -3,11 +3,14 @@
 > sass-lint -v -q
 
 
+src/Drawer/Drawer.scss
+  92:3  warning  Mixed tabs and spaces  indentation
```
Comment (talend-glorieux) [inline output/components.sasslint.txt]: Please fix that issue.

File: packages/components/src/Drawer/Drawer.scss [hunk 2]
```diff
@@ -46,15 +48,36 @@ $tc-action-bar-background-color: $concrete !default;
...
+}
+
 .tc-drawer-header {
-	padding: $padding-normal;
 	background-color: $brand-primary-lighter;
 
-	h1 {
+	.tc-drawer-header-title {
+		padding: $padding-normal;
+		position: relative;
```
Comment (frassinier) [inline packages/components/src/Drawer/Drawer.scss]: Why is this relative?

File: packages/components/src/Drawer/Drawer.scss [hunk 3]
```diff
@@ -46,15 +48,36 @@ $tc-action-bar-background-color: $concrete !default;
...
+		}
+	}
+}
+
 .tc-drawer-header {
-	padding: $padding-normal;
 	background-color: $brand-primary-lighter;
 
-	h1 {
+	.tc-drawer-header-title {
```
Comment (frassinier) [inline packages/components/src/Drawer/Drawer.scss]: Please, use nesting, DRY
```scss
.tc-drawer-header {
  &-title {
    /* ... */
  }
}
```

File: packages/components/src/Drawer/Drawer.scss [hunk 4]
```diff
@@ -46,15 +48,36 @@ $tc-action-bar-background-color: $concrete !default;
...
+		padding: $padding-normal;
+		position: relative;
+	}
+
+	.tc-drawer-header-title h1 {
```
Comment (frassinier) [inline packages/components/src/Drawer/Drawer.scss]: Please, use nesting, DRY
```scss 
.tc-drawer-header {
  &-title {
    h1 {
    }
  }
}
```

File: packages/components/src/Drawer/Drawer.component.js [hunk 5]
```diff
@@ -93,8 +93,8 @@ function DrawerTitle({ title, children, onCancelAction }) {
 		return null;
 	}
 	return (
-		<div>
-			<div className={theme['tc-drawer-header']}>
+		<div className={theme['tc-drawer-header']}>
+			<div className={theme['tc-drawer-header-title']}>
```
Comment (frassinier) [inline packages/components/src/Drawer/Drawer.component.js]: If we can have ellipsis, I suggest to deal with `title` attribute to display full content if needed.

File: packages/components/src/Drawer/Drawer.scss (hunk 4 above)
Comment (frassinier) [reply packages/components/src/Drawer/Drawer.scss]: This is optional

File: packages/components/src/Drawer/Drawer.scss (hunk 3 above)
Comment (frassinier) [reply packages/components/src/Drawer/Drawer.scss]: This is optional

File: packages/components/src/Drawer/Drawer.scss (hunk 2 above)
Comment (vyshkov) [reply packages/components/src/Drawer/Drawer.scss]: @frassinier because 'cancel' cross button has position absolute.

**Pull request merged**